test/PushComm/unit_tests/SubscribeBySig/SubscribeBySig.t.sol

```diff
@@ -0,0 +1,294 @@
+pragma solidity ^0.8.0;
+
+import { BaseTest } from "../../../BaseTest.t.sol";
+import { CoreTypes } from "../../../../contracts/libraries/DataTypes.sol";
+import { SignatureVerifier } from "contracts/mocks/MockERC1271.sol";
+
+import "forge-std/console.sol";
+
+contract SubscribeBySig_Test is BaseTest {
+    bytes constant _testChannelIdentity = bytes("test-channel-hello-world");
+
+    event Subscribe(address indexed channel, address indexed user);
+    event Unsubscribe(address indexed channel, address indexed user);
+
+    SignatureVerifier verifierContract;
+
+    function setUp() public override {
+        BaseTest.setUp();
+
+        changePrank(actor.tim_push_holder);
+        verifierContract = new SignatureVerifier();
+
+        changePrank(actor.bob_channel_owner);
+        coreProxy.createChannelWithPUSH(CoreTypes.ChannelType.InterestBearingOpen, _testChannelIdentity, 50e18, 0);
+    }
+
+    modifier whenUserSubscribesWith712Sig() {
+        _;
+    }
+
+    function testFail_WhenMaliciousUserReplaysA712Signature() public whenUserSubscribesWith712Sig {
+        bytes32 DOMAIN_SEPARATOR = getDomainSeparator();
+        SubscribeUnsubscribe memory _subscribeUnsubscribe = SubscribeUnsubscribe(
+            actor.bob_channel_owner,
+            actor.alice_channel_owner,
+            commProxy.nonces(actor.alice_channel_owner),
+            block.timestamp + 1000
+        );
+        bytes32 structHash = getStructHash(_subscribeUnsubscribe, SUBSCRIBE_TYPEHASH);
+        bytes32 digest = keccak256(abi.encodePacked("\x19\x01", DOMAIN_SEPARATOR, structHash));
+        (uint8 v, bytes32 r, bytes32 s) = vm.sign(privateKeys[actor.alice_channel_owner], digest);
+
+        commProxy.subscribeBySig(
+            actor.bob_channel_owner,
+            actor.alice_channel_owner,
+            commProxy.nonces(actor.alice_channel_owner),
+            block.timestamp + 1000,
+            v,
+            r,
+            s
+        );
+        uint256 oldNonce = commProxy.nonces(actor.alice_channel_owner) - 1;
+        changePrank(actor.charlie_channel_owner);
+        // Revert fails for unknown reason - Needs to be fixed
+        //vm.expectRevert(bytes("PushCommV2::subscribeBySig: Invalid nonce"));
+        commProxy.subscribeBySig(
+            actor.bob_channel_owner, actor.alice_channel_owner, oldNonce, block.timestamp + 1000, v, r, s
+        );
+    }
+
+    function test_WhenAnExpired712SignatureIsPassed() public whenUserSubscribesWith712Sig {
```

Comment:
as this is a test that has expectRevert, this should've been named as test_Revertwhen_
It's good to maintain uniformity across all tests

It's the case with few tests in this file, all should be changed accordingly